staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -721,17 +724,18 @@ func shouldDelegateList(opts storage.ListOptions) bool {
 	pred := opts.Predicate
 	match := opts.ResourceVersionMatch
 	pagingEnabled := utilfeature.DefaultFeatureGate.Enabled(features.APIListChunking)
+	consistentListFromCacheEnabled := utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache)
+
+	// Serve consistent reads from storage if ConsistentListFromCache is disabled
+	consistentReadFromStorage := resourceVersion == "" && !consistentListFromCacheEnabled
+	// Watch cache doesn't support continuations, so serve them from etcd.
 	hasContinuation := pagingEnabled && len(pred.Continue) > 0
+	// Serve paginated requests about revision "0" from watch cache to avoid overwhelming etcd.
 	hasLimit := pagingEnabled && pred.Limit > 0 && resourceVersion != "0"
+	// Watch cache only supports ResourceVersionMatchExact (default).
```

Comment:
it only supports NotOlderThan

FWIW - those kind of small clanups/refactor, can you please move to a separate PR?

Reply:
Same for test improvements that you did a couple that are useful even without this change.

Reply:
I could, if I knew it will not be bottlenecked on review :P

Reply:
https://github.com/kubernetes/kubernetes/pull/118792

Reply:
https://github.com/kubernetes/kubernetes/pull/118794

Reply:
Split 3 commits out of the original one. Created separate PR for each of them, but still left them in this PR. 

Leaving it up to you to either review the small or big PRs.